helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedResourceWeightsProvider.java

```diff
@@ -0,0 +1,49 @@
+package org.apache.helix.controller.rebalancer.waged;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.helix.controller.dataproviders.ResourceControllerDataProvider;
+import org.apache.helix.controller.rebalancer.util.WagedRebalanceUtil;
+import org.apache.helix.model.IdealState;
+import org.apache.helix.model.ResourceConfig;
+
+
+/**
+ * A weights data provider for waged resources.
+ */
+public class WagedResourceWeightsProvider {
```

Comment:
If i understand correctly then this is just merges ideal state and resource config and return weights for paritions right? If yes then do you think moving it to util method would help rather than having a separate class?

Reply:
Yeah, that's true, a util also works. I made it this way hoping this is more object-oriented, and to implement an interface https://github.com/apache/helix/pull/2463/files#diff-ae7e0f8fcf2cb9f7d514f6de739a3b9bb9e5a05cc5eb5e86c22fdbd8f221cf2e, and this is just one data provider for WAGED.